powerdns/serializers.py

```diff
@@ -75,6 +79,17 @@ class RecordSerializerV2(RecordSerializer):
         view_name='domain-detail',
     )
 
+    def validate(self, attrs):
+        domain, content, record_type = (
+            attrs.get('domain'), attrs.get('content'), attrs.get('type')
+        )
+        if domain and domain.is_public and content and record_type == 'A':
+            address = ipaddress.ip_address(content)
+            if address.is_private:
+                raise ValidationError('IP address can not be private.')
```

Comment:
I'm afraid that it's not that simple. AFAIK we have private records (ips) under public domains right now.